python/GafferSceneUI/InteractiveRenderUI.py

```diff
@@ -229,6 +229,45 @@ def __startPauseClicked( self, button ) :
 	def __stopClicked( self, button ) :
 		self.getPlug().setValue( GafferScene.InteractiveRender.State.Stopped )
 
+##########################################################################
+# UI for the messages plug that presents the render log
+##########################################################################
+
+class _MessagesPlugValueWidget( GafferUI.PlugValueWidget ) :
+
+	def __init__( self, plug, *args, **kwargs) :
+
+		self.__messages = GafferUI.MessageWidget(
+			toolbars = True,
+			follow = True,
+			role = GafferUI.MessageWidget.Role.Log
+		)
+
+		GafferUI.PlugValueWidget.__init__( self, self.__messages, plug )
+
+		self.visibilityChangedSignal().connect( Gaffer.WeakMethod( self._updateFromPlug ), scoped = False )
```

Comment:
How important is the dont-update-when-invisible optimisation? Rather than implement it manually, perhaps we should use `@LazyMethod`, both for simplicity and also to better communicate intent?

Reply:
So much better, I hadn't realised how neat `@LazyMethod` was :) https://github.com/GafferHQ/gaffer/pull/3790/commits/9317c15f9b01b9b59352d415896ab03f753607de